src/socket/address.cr

```diff
@@ -227,7 +227,10 @@ class Socket
       in LibC::InAddr
         addr.s_addr & 0x000000ff_u32 == 0x0000007f_u32
       in LibC::In6Addr
-        ipv6_addr8(addr) == StaticArray[0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 1_u8]
+        addr8 = ipv6_addr8(addr)
+        num = addr8.unsafe_as(UInt128)
+        num == (1_u128 << 120) ||                                                      # "::1"
+          num & 0xffffffffffffffffffffffffff_u128 == 0x7fffff00000000000000000000_u128 # "::ffff:127.0.0.1/104"
```

Comment:
Those big UInt128 literals are not supported on older Crystal versions, we should avoid them for now (see #5545).
```suggestion
          # TODO: Use UInt128 literals
          num & UInt128::MAX >> 24 == 0x7fffff_u128 << 80 # "::ffff:127.0.0.1/104"
```